integreat_cms/api/v3/socialmedia_headers.py

```diff
@@ -0,0 +1,298 @@
+"""
+This module includes functions related to the social media headers API endpoint.
+"""
+from __future__ import annotations
+
+import logging
+from html import unescape
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+from django.shortcuts import render
+from django.utils.html import strip_tags
+from django.utils.translation import activate
+
+from integreat_cms.cms.models.events.event_translation import EventTranslation
+from integreat_cms.cms.models.languages.language import Language
+from integreat_cms.cms.models.pois.poi_translation import POITranslation
+from integreat_cms.cms.models.push_notifications.push_notification_translation import (
+    PushNotificationTranslation,
+)
+from integreat_cms.cms.models.regions.region import Region
+
+from ...cms.models import PageTranslation
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from django.http import HttpRequest, HttpResponse
+
+logger = logging.getLogger(__name__)
+
+
+def socialmedia_headers(
+    request: HttpRequest, path: Optional[str] = None
```

Comment:
```suggestion
    from django.http import HttpRequest, HttpResponse

logger = logging.getLogger(__name__)


def socialmedia_headers(
    request: HttpRequest, path: str | None = None
```
Although the python world [doesn't completely agree](https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945) on one consistent syntax, the `|` union separator seems to be the more modern way to express `Union` and `Optional` (see [PEP 604](https://peps.python.org/pep-0604/)).